src/test-run/index.js

```diff
@@ -199,34 +235,19 @@ ServiceMessages[CLIENT_MESSAGES.ready] = function (msg) {
 
     this.pendingRequest = null;
 
-    if (!this.running)
-        this._start();
-
-    var driverStatus                  = msg.status;
-    var shouldRejectPendingDriverTask = driverStatus.pageError &&
-                                        this.pendingDriverTask &&
-                                        isCommandRejectableByPageError(this.pendingDriverTask.command);
+    var driverStatus   = msg.status;
+    var cachedResponse = this.lastDriverStatusId === driverStatus.id ? this.lastDriverStatusResponse : null;
```

Comment:
why we need this var? Can't we just do:
```js
if(msg.status.id === this.lastDriverStatusId)
     return this.cachedResponse;
```

AFAIK, we don't need to return promise from the handler BTW.